saga/sagalog.go

```diff
@@ -34,6 +38,23 @@ type SagaLog interface {
 	GetActiveSagas() ([]string, error)
 }
 
+// CorruptedSagaLogError this is a critical error specifies
+// that the data stored in the sagalog for a specified saga
+// is corrupted and unrecoverable.
+type CorruptedSagaLogError struct {
+	s string
+}
+
+func (e CorruptedSagaLogError) Error() string {
+	return e.s
+}
+
+func NewCorruptedSagaLogError(sagaId string, msg string) error {
+	return CorruptedSagaLogError{
```

Comment:
Might just want to store sagaID and msg as fields in CorruptedSagaLogError, and then do the Sprintf in the call to Error(). This could make it easier to handle the data programatically?